cpp/src/randomforest/randomforest.cu

```diff
@@ -308,53 +310,49 @@ std::string get_rf_json(const RandomForestMetaData<T, L>* forest) {
 }
 
 template <class T, class L>
-void build_treelite_forest(ModelHandle* model,
+void build_treelite_forest(ModelHandle* model_handle,
                            const RandomForestMetaData<T, L>* forest,
                            int num_features, int task_category) {
-  // Non-zero value here for random forest models.
-  // The value should be set to 0 if the model is gradient boosted trees.
-  int random_forest_flag = 1;
-  ModelBuilderHandle model_builder;
-  // num_class is 1 for binary classification and regression
-  // num_class is #class for multiclass classification which is the same as task_category
-  int num_class = task_category > 2 ? task_category : 1;
-
-  const char* leaf_type = DecisionTree::TreeliteType<L>::value;
-  if (std::is_same<L, int>::value) {
-    // Treelite codegen doesn't yet support integer leaf output
-    leaf_type = DecisionTree::TreeliteType<float>::value;
+  auto parent_model = tl::Model::Create<T, T>();
+  tl::ModelImpl<T, T>* model =
+    dynamic_cast<tl::ModelImpl<T, T>*>(parent_model.get());
+  if (model == nullptr) {
+    throw raft::exception("Invalid downcast to Treelite ModelImpl");
   }
 
-  TREELITE_CHECK(TreeliteCreateModelBuilder(
-    num_features, num_class, random_forest_flag,
-    DecisionTree::TreeliteType<T>::value, leaf_type, &model_builder));
-
+  unsigned int num_class;
   if (task_category > 2) {
     // Multi-class classification
-    TREELITE_CHECK(TreeliteModelBuilderSetModelParam(
-      model_builder, "pred_transform", "max_index"));
+    num_class = task_category;
+    model->task_type = tl::TaskType::kMultiClfProbDistLeaf;
+    model->task_param = tl::TaskParameter{tl::TaskParameter::OutputType::kFloat,
```

Comment:
This initialization expression for `task_param` works in both cases (multi-class and binary/regression), so consider placing it after the conditional.

Reply:
They're slightly different. Note the final argument in each.

Reply:
`num_class` as the final argument should work in the 2nd case as well (`num_class == 1` in that case).

Reply:
Ah, right you are!